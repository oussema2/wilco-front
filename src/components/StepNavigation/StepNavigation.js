import React, { useEffect, useRef } from 'react';
import { View, ViewPropTypes } from 'react-native';
import PropTypes from 'prop-types';
import PagerView from 'react-native-pager-view';
import { styles } from './StepNavigation.styles';
import StepIndicator from '../../screens/SignUp/Components/StepIndicator';
import { NavigationBar } from '../NavigationBar';
import noop from '../../helpers/noop';
import { HorizontalPadding } from '../HorizontalPadding';

const StepNavigation = ( {
	currentStep, children, style, onBackButtonPress
} ) => {
	const viewPagerRef = useRef( undefined );
	const stepCount = children.length;

	useEffect( () => {
		viewPagerRef.current.setPage( currentStep );
	}, [ currentStep ] );

	return (
		<View style={[ styles.StepNavigation, style ]}>
			<HorizontalPadding>
				<NavigationBar
					testID="navigationBar-component"
					onBackArrowPress={onBackButtonPress}
					titleComponent={<StepIndicator currentStep={currentStep} stepCount={stepCount} />}
				/>
			</HorizontalPadding>
			<PagerView
				ref={viewPagerRef}
				style={styles.viewPager}
				initialPage={0}
				scrollEnabled={false}
			>

				{children}
			</PagerView>
		</View>
	);
};

StepNavigation.propTypes = {
	children: PropTypes.node,
	currentStep: PropTypes.number,
	style: ViewPropTypes.style,
	onBackButtonPress: PropTypes.func
};

StepNavigation.defaultProps = {
	children: [],
	currentStep: 0,
	style: {},
	onBackButtonPress: noop
};

export default StepNavigation;
