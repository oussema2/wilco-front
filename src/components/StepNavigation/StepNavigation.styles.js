import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create( {
	StepNavigation: {
		flex: 1
	},
	viewPager: {
		flex: 1
	}
} );
