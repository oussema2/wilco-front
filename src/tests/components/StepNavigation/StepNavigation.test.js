import * as React from 'react';
import { render, cleanup } from '@testing-library/react-native';
import { Text, View } from 'react-native';
import StepNavigation from '../../../components/StepNavigation/StepNavigation';

afterEach( cleanup );

describe( 'StepNavigation', () => {
	const testID = 'StepNavigation-testID';

	describe( 'with the default props', () => {
		it( 'renders the StepNavigation component correctly', () => {
			const component = render(
				<StepNavigation
					testID={testID}
					currentStep={0}
				>
					<View testID="step_one"><Text>Step one</Text></View>
					<View testID="step_two"><Text>Step two</Text></View>
				</StepNavigation>
			);

			expect( component ).toMatchSnapshot();
		} );
	} );
} );
