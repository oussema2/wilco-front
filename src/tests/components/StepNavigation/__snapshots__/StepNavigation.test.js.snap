// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepNavigation with the default props renders the StepNavigation component correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {},
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "paddingHorizontal": 20,
        },
        null,
      ]
    }
    testID="horizontalPadding-component"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "minHeight": 44,
        }
      }
      testID="navigationBar-component"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "center",
              "minHeight": 44,
            },
            Object {
              "flex": 0.25,
              "justifyContent": "flex-start",
            },
          ]
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 15,
              "left": 15,
              "right": 15,
              "top": 15,
            }
          }
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/icons/chevronLeft.png",
              }
            }
            style={
              Object {
                "height": 20,
                "width": 20,
              }
            }
            testID="backArrow-image"
          />
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "center",
              "minHeight": 44,
            },
            Object {
              "flex": 0.5,
              "justifyContent": "center",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
          testID="FirstStep-component"
        >
          <View
            style={
              Object {
                "backgroundColor": "#12A3BE",
                "borderRadius": 5,
                "height": 8,
                "marginRight": 8,
                "width": 24,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "#D3D3D3",
                "borderRadius": 5,
                "height": 8,
                "marginRight": 8,
                "width": 24,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "center",
              "minHeight": 44,
            },
            Object {
              "flex": 0.25,
              "justifyContent": "flex-end",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "width": 20,
            }
          }
          testID="rightPlaceholder-view"
        />
      </View>
    </View>
  </View>
  <RNCViewPager
    layoutDirection="ltr"
    onMoveShouldSetResponderCapture={[Function]}
    onPageScroll={[Function]}
    onPageScrollStateChanged={[Function]}
    onPageSelected={[Function]}
  >
    <View
      collapsable={false}
      style={
        Array [
          undefined,
          Object {
            "bottom": 0,
            "height": undefined,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": undefined,
          },
        ]
      }
      testID="step_one"
    >
      <Text>
        Step one
      </Text>
    </View>
    <View
      collapsable={false}
      style={
        Array [
          undefined,
          Object {
            "bottom": 0,
            "height": undefined,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": undefined,
          },
        ]
      }
      testID="step_two"
    >
      <Text>
        Step two
      </Text>
    </View>
  </RNCViewPager>
</View>
`;
